### PR TITLE
Backport JITServer chtable related changes from master

### PR DESCRIPTION
**Pull request by harryyu1994**
Backport #7113 , had to bring #6932 along with it as it contains chtable changes. Also to make the next rebase easier.

Signed-off-by: Harry Yu <harryyu1994@gmail.com>

### CODE REVIEW

**Comment by harryyu1994** [comment]
https://github.com/eclipse/openj9/pull/6999
We need to port `IProfiler` changes first before porting `JITServerIProfiler` changes.

**Comment by harryyu1994** [comment]
@mpirvu Hi Marius, could you help review this change?

File: runtime/compiler/env/JITServerPersistentCHTable.cpp [hunk 1]
```diff
@@ -149,9 +141,7 @@ JITServerPersistentCHTable::commitModifications(const std::string &rawData)
          {
          auto classInfo = findClassInfo(flat->_subClasses[i]);
 
-         // For some reason the subclass info is still null sometimes. This may be indicative of a larger problem or it could be harmless.
-         // JITServer TODO: figure out why this is happening
-         //TR_ASSERT(classInfo, "subclass info cannot be null: ensure subclasses are loaded before superclass");
+         TR_ASSERT_FATAL(classInfo, "subclass info cannot be null: ensure subclasses are loaded before superclass");
```
Comment (mpirvu) [inline runtime/compiler/env/JITServerPersistentCHTable.cpp]: Let's make this into a TR_ASSERT temporarily. The right solution is TR_ASSERT_FATAL, but currently there is a known issue which will hit this assert, the server will die and we will not test the right thing. I am hoping to get to this problem soon and then we can put back the ASSERT_FATAL before the final sync between master and jitaas.

**Comment by mpirvu** [comment]
Jenkins test sanity.functional+jitaas xlinux jdk8,jdk11 depends eclipse/openj9-omr#jitaas

**Comment by mpirvu** [comment]
Only the known `cmdLineTester_gcsuballoctests_0` test failed, hence merging.

**Pull request merged**